torch_xla/csrc/ops/ops.cpp

```diff
@@ -863,6 +864,38 @@ NodePtr LogicalOr(const Value& input, const Value& other) {
       std::move(lower_fn));
 }
 
+NodePtr NanToNum(const Value& input, const at::Scalar& nan,
+                 const at::Scalar& posinf, const at::Scalar& neginf) {
```

Comment:
The better way is to call `XLATensor::GetIrValueForScalar` for  `nan`, `posinf` and `neginf `, pass them as IR::Value to the Node here. You don't need to worry about the Hash function here too if you do that since all IR::Value will be taken care of.

The problem with your current approach is that hardcoding the `at::Scalar` as `xla::Op` will result in a recompilation every-time the scalar value changed, which is not ideal. Calling `GetIrValueForScalar` early will create a `device_data` for all three scalars and perform a host->device upload. This way the graph will not has any assumption with the actual value of these scalars and won't recompile when they changed.